pkg/controller/deployment/deployment_controller.go

```diff
@@ -375,6 +388,14 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return nil
 	}
 	d := *obj.(*extensions.Deployment)
+	if !dc.rcStoreSynced() {
```

Comment:
Do we need to wait for the whole store to sync!
What if just check that old and new RCs dont have any unsatisfied expectation?